src/storage/src/source/postgres.rs

```diff
@@ -1024,6 +1025,14 @@ async fn produce_replication<'a>(
 
             // The inner loop
             loop {
+                use LogicalReplicationMessage::*;
+                metrics.total.inc();
+
+                // Wait no longer than `FEEDBACK_INTERVAL` on the stream; if we exceed the deadline,
+                // we should let the upstream PG source know we're still here. This also gives us an
+                // opportunity to check that it's still alive.
+                let res = tokio::time::timeout(FEEDBACK_INTERVAL, stream.as_mut().next()).await;
```

Comment:
I think in theory this should be `FEEDBACK_INTERVAL.saturating_sub(last_feedback.elapsed())`? But let's roll with what you have for now, since I'd like to backport it into v0.39.x.